document-portal/xdp-main.c

```diff
@@ -443,7 +443,7 @@ validate_fd (int fd,
   if (app_id != NULL && *app_id != 0 &&
       g_str_has_prefix (path_buffer, "/newroot/app/"))
     {
-      char *rel_path = path_buffer + strlen ("/newroot/app/");
+      char *rel_path = &path_buffer[13];
```

Comment:
Please use strlen here. It will optimize to the same thing, but its much easier to read/debug